README.md

```diff
@@ -32,39 +32,26 @@ The binaries are available as a [NuGet package](https://www.nuget.org/packages/n
 
 For all information about MQTT protocol, please visit official web site  http://mqtt.org/. It is recommended to have a good understanding of how MQTT protocol is working to properly use it. The mechanism of Quality of Service is an important one to understand.
 
-## Usage
+### Examples
 
-The usage is globally the same whatever version is used. There are some specificities between v3.1.1 and v5.0. The version 5.0 brings more control and additional properties. For convenience, they are all commented with `v5.0 only` in the properties comments. If you're using a v5.0 property with the v3.1 or v3.1.1 protocol, they'll just be ignored.
+The M2Mqtt library provides a main class `MqttClient` that represents the MQTT client to connect to a broker. You can connect to the broker providing its IP address or host name and optionally some parameters related to MQTT protocol.
 
-Here is a basic example of creating a v3.1.1 server and connecting to it:
+After connecting to the broker you can use `Publish()` method to publish a message to a topic and `Subscribe()` method to subscribe to a topic and receive message published on it. The `MqttClient` class is events based so that you receive an event when a message is published to a topic you subscribed to. You can receive event when a message publishing is complete, you have subscribed or unsubscribed to a topic.
 
-```csharp
-MqttClient mqtt = new MqttClient("test.mosquitto.org", 8883, true, new X509Certificate(CertMosquitto), null, MqttSslProtocols.TLSv1_2);
-var ret = mqtt.Connect("nanoTestDevice", true);
-if (ret != MqttReasonCode.Success)
-{
-    Debug.WriteLine($"ERROR connecting: {ret}");
-    mqtt.Disconnect();
-    return;
-}
-```
+For more details and samples, go to the [main Samples repo](https://github.com/nanoframework/Samples/MQTT).
+
+## MQTT v5 Usage
+
+The usage is globally the same whatever version is used. There are some specificities between v3.1.1 and v5.0. The version 5.0 brings more control and additional properties. For convenience, they are all commented with `v5.0 only` in the properties comments. If you're using a v5.0 property with the v3.1 or v3.1.1 protocol, they'll just be ignored.
 
-For the v5.0, you just need to specify the version before the connection:
+For the v5.0, you just need to specify the version before the connection, like this:
 
 ```csharp
-MqttClient mqtt = new MqttClient("test.mosquitto.org", 8883, true, new X509Certificate(CertMosquitto), null, MqttSslProtocols.TLSv1_2);
```

Comment:
if you remove this, it won't work. Please don't remove code that will break the way the secure server is working.

Reply:
Which code won't work? And why the code that I left won't work?